lib/documents.js

```diff
@@ -252,6 +282,30 @@ export async function getRegistration({internalId} = {}) {
   return record;
 }
 
+/**
+ * Retrieves a registration record by `externalId`.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.externalIdHash - The external ID hash.
+ *
+ * @returns {Promise<object>} The registration record.
+ */
+export async function getRegistrationbyExternalId({externalId}) {
```

Comment:
This method is problematic. There can be multiple registration records for a given `externalId`, they are only unique for a combination of `externalIdHash` and `documentHash`. If we *have* to expose a call to get a registration, it will need both the external ID and the document. But I suspect we don't even want to expose such a thing -- because it likely has race conditions with its usage.

Reply:
Removed.